notebooks/ch17/classification_multiclass.md

```diff
@@ -0,0 +1,312 @@
+      <td>1.4</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>4.7</td>
+      <td>3.2</td>
+      <td>1.3</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>4.6</td>
+      <td>3.1</td>
+      <td>1.5</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>5.0</td>
+      <td>3.6</td>
+      <td>1.4</td>
+      <td>0.2</td>
+      <td>Iris-setosa</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+X, y = iris.drop('species', axis=1), iris['species']
+```
+
+
+```python
+# Alternatively, read in from sklearn.datasets
+# from sklearn.datasets import load_iris
+
+# X, y = load_iris(return_X_y=True)
+```
+
+
+```python
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.35, random_state=42)
+```
+
+After dividing the dataset into train and test splits, we fit a multiclass classifier to our training data. By default, scikit-learn's `LogisticRegression` sets `multi_class='ovr'`:
+
+
+```python
+lr = LogisticRegression(random_state=42)
+lr.fit(X_train, y_train)
+```
+
+
+
+
+    LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
+              intercept_scaling=1, max_iter=100, multi_class='ovr', n_jobs=1,
+              penalty='l2', random_state=42, solver='liblinear', tol=0.0001,
+              verbose=0, warm_start=False)
+
+
+
+We predict on the test data, and use a confusion matrix to evaluate the results:
+
+
+```python
+# to-do: add x-axis, y-axis labels to confusion matrix
+y_pred = lr.predict(X_test)
+confusion_matrix(y_test, y_pred)
+```
+
+
+
+
+    array([[19,  0,  0],
+           [ 0, 15,  2],
+           [ 0,  0, 17]])
+
+
+
+The confusion matrix shows that our classifier misclassified two `Iris-versicolour` observations as `Iris-virginica`. In the real world, certain misclassifications can be more common than others. Confusion matrices are valuable because they help us identify where our classifier struggles, and thus provides insight on what kind of features we may need to extract in order to improve the classifier.
+
+## Multilabel Classification
+
+Another type of classification problem is **multilabel classification**, in which each observation can have multiple labels. An example would be a document classification system: a document can be classified as having positive or negative sentiments, but it can also be distinguished between religious/nonreligious or political/apolitical. Multilabel problems can also be multiclass; we may want our document classification system to distinguish between a list of genres, as well as identify the language that the document is written in. 
+
+Because each set of labels in a multiclass classification is mutually exclusive, each problem can be solved independently of the others. We can then combine the outputs so that every observation is assigned a set of classification labels.
```

Comment:
(Just adding to Sam's comment here): Consider the definition given here also http://scikit-learn.org/stable/modules/multiclass.html